src/repos.cc

```diff
@@ -1782,6 +1805,11 @@ void add_repo_by_url( Zypper & zypper,
     repo.setName(it->second.front());
   repo.addBaseUrl(url);
 
+  long long prio = parse_priority(zypper);
```

Comment:
```unsigned prio = parse_priority(zypper);``` 
Parsing (as it may throw) should be moved to the beginning of add_repo_by_url; e.g. before ```RepoInfo repo;```.